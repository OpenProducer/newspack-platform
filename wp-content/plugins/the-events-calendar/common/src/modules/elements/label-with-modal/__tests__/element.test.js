/**
 * External dependencies
 */
import React from 'react';

/**
 * Internal dependencies
 */
import LabelWithModal from '@moderntribe/common/elements/label-with-modal/element';

describe( 'Label With Modal Element', () => {
	it( 'renders a label with modal', () => {
		const component = renderer.create( <LabelWithModal /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders a label with modal with class', () => {
		const component = renderer.create( <LabelWithModal className="test-class" /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders a label with modal with label', () => {
		const component = renderer.create( <LabelWithModal label="test label" /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
