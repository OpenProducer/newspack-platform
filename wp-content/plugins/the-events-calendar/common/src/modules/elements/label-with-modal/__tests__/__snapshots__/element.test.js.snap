// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label With Modal Element renders a label with modal 1`] = `
<div
  className="tribe-editor__labeled-item tribe-editor__label-with-modal"
>
  <span
    className="tribe-editor__labeled-item__label"
  />
  <div
    className="tribe-editor__modal-button tribe-editor__label-with-modal__modal-button"
  >
    <button
      className="tribe-editor__button tribe-editor__modal-button__button"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Label With Modal Element renders a label with modal with class 1`] = `
<div
  className="tribe-editor__labeled-item tribe-editor__label-with-modal test-class"
>
  <span
    className="tribe-editor__labeled-item__label"
  />
  <div
    className="tribe-editor__modal-button tribe-editor__label-with-modal__modal-button"
  >
    <button
      className="tribe-editor__button tribe-editor__modal-button__button"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Label With Modal Element renders a label with modal with label 1`] = `
<div
  className="tribe-editor__labeled-item tribe-editor__label-with-modal"
>
  <span
    className="tribe-editor__labeled-item__label"
  >
    test label
  </span>
  <div
    className="tribe-editor__modal-button tribe-editor__label-with-modal__modal-button"
  >
    <button
      className="tribe-editor__button tribe-editor__modal-button__button"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
